### ISSUE TITLE
Advanced Shortcut - default main menu shortcuts not working

### ISSUE THREAD

**Issue by gene-pavlovsky** [fdorg/flashdevelop#1302]
Originally reported in [here](https://github.com/fdorg/flashdevelop/pull/1245#issuecomment-242407347)
E.g. `Alt+T` should open `Tools` menu, instead statusbar says "Alt+T was pressed, waiting for second key...".

**Comment by Neverbirth** [comment]
@wise0704 The reason is "simple", command keys have priority over mnemonics, and they are separate concepts, hence ProcessCmdKey has to return false in order for a mnemonic to be processed.

I've checked VS, by default it doesn't have any list of possible "valid first extended shortcut", they list is built by the key mapping the user has, ie.: by default Alt+T will open the Tools menu, when the user sets an advanced shortcut starting with Alt+T, this would return true in `ShortcutKeysManager.IsValidExtendedShortcutFirst` and the Tools menu will stop working because it will start waiting for the second part of the command.

What do you think, could you do this soon? For me this one is not a top priority at the moment.

**Comment by wise0704** [comment]
I will look again into the .NET shortcut system soon.

**Comment by wise0704** [comment]
By shortcuts, I meant keyboard inputs.
When I was doing the PR, I ran through the decompiled assemblies. The mnemonics are handled in the `ToolStripManager` iirc, which is called in the `ProcessCmdKey()` in the root `Form`. 
Remember, at least for you and me, the mnemonics still work with the changes. There should be some other factor.

**Comment by Neverbirth** [comment]
Yeah, I know what you meant, but for the .NET guys shortcuts were what's processed in ProcessCmdKey, that's why I said "not".

Again, mnemonics are not handled in there (after ProcessCmdKey returns false, after IsInputChar returns false, and other checks "fail" too), and any control may have them, so I don't think ToolStripManager gets a hold of all of them? (could be, but feels strange?). Forgot to say, for me mnemonics were working, but because I had some custom changes, my bad! with baseline development branch they wouldn't work.

**Comment by Neverbirth** [comment]
Played with and checked a bit default keyboard implementation, and it doesn't seem that `ProcessCmdKey `checks mnemonics at all, in the default implementation if there is a shortcut and a mnemonic using the same combination, the shortcut will always have the preference (I played using toolstrips, normal controls, using different tab orders, etc, etc). It was as I was expecting.

The one I was not expecting: if you have a menustrip with `Text`, for example, Main, and a button with `Text `"&MTest", the button Click event will run as expected, however, if you rename the menu to "Main", the strip will open, so it seems the menustrip consumes the key before it reaches `ProcessMnemonic`. Nothing really important to our case, but to me it seems like a Windows Forms flaw.

**Comment by wise0704** [comment]
I can confirm it doesn't work now that I tried it. I don't know when it still worked (would've been during the development stages until some commit).

While working on this, I found something strange.
When I press `Alt+T` to open the Tools menu, and then use the mouse to cancel instead of pressing `Alt` again, the underlines do disappear; however when I press `Alt` again to show the underlines, it opens up the first menu (i.e. File) as well, thus causing what @Neverbirth described before, that `Alt+T` could restart FD in this case.
